shaders/commonProfileShaders.cpp

```diff
@@ -78,70 +85,74 @@ namespace GLTF
+                        png_uint_32 color_type = png_get_color_type(pngPtr, infoPtr);
+                        if (color_type == PNG_COLOR_TYPE_RGB_ALPHA || color_type == PNG_COLOR_TYPE_GRAY_ALPHA) {
+                            int width = png_get_image_width(pngPtr, infoPtr);
+                            int height = png_get_image_height(pngPtr, infoPtr);
+                            png_byte bit_depth = png_get_bit_depth(pngPtr, infoPtr);
+    
+                            if (bit_depth == 16)
+                                png_set_strip_16(pngPtr);
+    
+                            if (png_get_valid(pngPtr, infoPtr, PNG_INFO_tRNS))
+                                png_set_tRNS_to_alpha(pngPtr);
+    
+                            if (color_type == PNG_COLOR_TYPE_GRAY_ALPHA)
+                                png_set_gray_to_rgb(pngPtr);
+    
+                            png_read_update_info(pngPtr, infoPtr);
+    
+                            // Allocate row pointers
+                            png_bytep *row_pointers = (png_bytep*)malloc(sizeof(png_bytep) * height);
+                            png_size_t rowBytes = png_get_rowbytes(pngPtr, infoPtr);
+                            for (int y = 0; y < height; y++) {
+                                row_pointers[y] = (png_byte*)malloc(rowBytes);
+                            }
+    
+                            png_read_image(pngPtr, row_pointers);
+    
+                            // Check for alpha less than 255
+                            for (int y = 0; (y < height) && !hasAlpha; y++) {
+                                png_bytep row = row_pointers[y];
+                                for (int x = 0; (x < width) && !hasAlpha; x++) {
+                                    png_bytep px = &(row[x * 4]);
+                                    if (px[3] != 255) {
+                                        hasAlpha = true;
+                                    }
                                 }
                             }
+    
+                            // Free row pointers
+                            for (int y = 0; y < height; y++) {
+                                free(row_pointers[y]);
+                            }
+                            free(row_pointers);
                         }
-
-                        // Free row pointers
-                        for (int y = 0; y < height; y++) {
-                            free(row_pointers[y]);
-                        }
-                        free(row_pointers);
+                        
+                        png_destroy_read_struct(&pngPtr, (png_infopp)0, (png_infopp)0);
                     }
-                    
-                    png_destroy_read_struct(&pngPtr, (png_infopp)0, (png_infopp)0);
                 }
             }
         }
-        
-        return hasAlpha;
 #else
-		return false;
+		hasAlpha = false;
 #endif
+
+		// For the love of sanity, cache this!
```

Comment:
I appreciate the spirit of this comment, but could you please make this solely descriptive? Just "cache the result" would be fine.